garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,247 @@ local meta = FindMetaTable( "Vector" )
 -- Nothing in here, still leaving this file here just in case
 
 --[[---------------------------------------------------------
-	Converts Vector To Color - alpha precision lost, must reset
+Converts Vector to Color - alpha precision lost, must reset
 -----------------------------------------------------------]]
-function meta:ToColor( )
+function meta:ToColor()
+	local x, y, z = self:Unpack()
+	return Color(x * 255, y * 255, z * 255)
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Table - removes all metrods from the copy
+-----------------------------------------------------------]]
+function meta:ToTable(arr)
+	local x, y, z = self:Unpack()
+	return {x = x, y = y, z = z}
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Array - removes all metrods from the copy
+-----------------------------------------------------------]]
+function meta:ToArray()
+	local x, y, z = self:Unpack()
+	return {x, y, z}
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Angle - data is copied
+-----------------------------------------------------------]]
+function meta:ToAngle()
+	local x, y, z = self:Unpack()
+	return Angle(x, y, z)
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a rotated vector. Kind of like v:GetNormalized()
+-----------------------------------------------------------]]
+function meta:GetRotated(...)
+	local v = Vector(self)
+	v:Rotate(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a multiplied vector
+-----------------------------------------------------------]]
+function meta:GetMul(...)
+	local v = Vector(self)
+	v:Mul(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of an added vector
+-----------------------------------------------------------]]
+function meta:GetAdd(...)
+	local v = Vector(self)
+	v:Add(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a vector
+-----------------------------------------------------------]]
+function meta:AddEx(x, y, z)
+	self[1] = self[1] + x

```

Comment:
I think ``*Ex`` should be changed to ``*Unpacked`` for naming consistency. Also use ``Unpack`` and ``SetUnpacked`` here.

Reply:
Ok !

Reply:
Done with Add/Sub*